### PR TITLE
Return proposals for current user

### LINKED ISSUE #901: Only return `proposals` that the current user is allowed to view
Similar to #759, we want the API to only return proposal data that the requesting user is authorized to see.

### PR DESCRIPTION
**Pull request by slifty**
This PR adds authorization to our `GET /proposals` endpoint.

Resolves #901

### CODE REVIEW

File: src/handlers/proposalsHandlers.ts [hunk 1]
```diff
@@ -22,10 +22,15 @@ import {
 import type { Request, Response, NextFunction } from 'express';
 
 const getProposals = (
-	req: Request,
+	req: AuthenticatedRequest,
 	res: Response,
 	next: NextFunction,
 ): void => {
+	if (req.user === undefined) {
+		next(new FailedMiddlewareError('Unexpected lack of user context.'));
```
Comment (reefdog) [inline src/handlers/proposalsHandlers.ts:30]: Tell me about it. _[shakes head]_ You and me both pal. _[raspberries in exasperation]_ What a day amiright

Comment (slifty) [reply src/handlers/proposalsHandlers.ts:30]: `Unexpected lack of joke.`

**Review by reefdog — APPROVED** [review]
This all looks good and is approved!

This approach doesn't lock us out of being able to *expand* scope for certain user types to see more than "their own" proposals, right? (Superadmins, e.g.)

**Comment by slifty** [comment]
@reefdog correct!  We'll need to implement that functionality which might result in changes to the query (or might result in a check in the middleware to pass `0` in / bypass the query in the event of a superuser).

**Pull request merged**